Tests/Tracking/CustomerIOIntegrationTests.swift

```diff
@@ -116,4 +116,43 @@ class CustomerIOIntegrationTests: IntegrationTest {
         XCTAssertGreaterThan(httpRequestRunnerStub.requestCallsCount, 0)
         XCTAssertEqual(diGraph.queueStorage.getInventory().count, 0)
     }
+
+    // MARK: Background queue behavior
+
+    // Adding of tasks to the SDK's background queue is performed on an OS background thread using an OS provided queue data structure. It's possible to configure the OS to run these background threads concurrently instead of serially which would add tasks to the background queue in a random order. This could cause HTTP errors as our background queue expects all tasks are in a specific order of events.
+    // This test sends lots of requests to the SDK and then we verify that all tasks added to the background queue are in order.
+    func test_backgroundQueue_givenAddManyRequestsToSDK_expectAddAllRequestsToBackgroundQueueInOrder() {
+        let numberOfTasksToAdd = 500
+
+        setUp { config in
+            // disable the background queue from executing so we don't need to stub HTTP requests in this test. This test is just testing adding tasks to the queue, not executing tasks in the queue.
+            config.backgroundQueueMinNumberOfTasks = numberOfTasksToAdd + 1
+        }
+
+        for index in 0 ..< numberOfTasksToAdd {
+            // Using trackMetric since it does not have requirements such as a profile identified to the SDK.
+            // Use loop index for the data to make it easy to verify queue tasks added in order.
+            CustomerIO.shared.trackMetric(deliveryID: String(index), event: .opened, deviceToken: String(index))
```

Comment:
unclear to me how this test shows that adding tasks is not happening on the main thread? to me it looks more like showing that the order is preserved.

Reply:
This test is not intended to test what thread the code is executed on. The test function is intending to test that when calling `CustomerIO.X` many times in a short amount of time, the requests are all adding to the BQ in the order requested by the customer.